src/gtk/toplevel.cpp

```diff
@@ -1282,12 +1282,18 @@ void wxTopLevelWindowGTK::DoMoveWindow(int WXUNUSED(x), int WXUNUSED(y), int WXU
 // window geometry
 // ----------------------------------------------------------------------------
 
-void wxTopLevelWindowGTK::GTKDoGetSize(int *width, int *height) const
+void
+wxTopLevelWindowGTK::GTKDoGetSize(bool isResizeable,
+                                  int *width, int *height) const
 {
     wxSize size(m_width, m_height);
-    size.x -= m_decorSize.left + m_decorSize.right;
-    size.y -= m_decorSize.top + m_decorSize.bottom;
-    size.IncTo(wxSize(0, 0));
+
+    if ( isResizeable || !HasClientDecor(m_widget) )
```

Comment:
BTW, does anybody (meaning @paulcor probably) know why do we check for not using CSD only for non-resizeable windows? I'd like to add a comment here explaining why do we do this, but I have absolutely no idea...

Reply:
8f47d9ad48d3c59090d3d6f8c9c4bb4eab9dce0f